client/src/components/Draftail/CommentableEditor/CommentableEditor.js

```diff
@@ -0,0 +1,217 @@
+import { DraftailEditor, ToolbarButton, createEditorStateFromRaw, serialiseEditorStateToRaw } from 'draftail';
+import { EditorState, Modifier, RichUtils, SelectionState } from 'draft-js';
+import { filterInlineStyles } from "draftjs-filters";
+import { useEffect, useMemo, useRef, useState } from 'react';
+import { useSelector, shallowEqual } from 'react-redux';
+
+import { STRINGS } from '../../../config/wagtailConfig';
+import Icon from '../../Icon/Icon';
+
+const COMMENT_STYLE_IDENTIFIER = 'COMMENT-'
+
+function usePrevious(value) {
+  const ref = useRef();
+  useEffect(() => {
+    ref.current = value;
+  }, [value]); // Only re-run if value changes
+  return ref.current;
+}
+
+class DraftailInlineAnnotation {
+    constructor(field) {
+      this.field = field;
+      this.inlineRefs = new Set()
+    }
+    addRef(ref) {
+      this.inlineRefs.add(ref);
+    }
+    removeRef(ref) {
+      this.inlineRefs.delete(ref);
+    }
+    getDesiredPosition() {
+      const nodeTops = Array.from(this.inlineRefs).map(ref => ref.current).filter(node => node).map(node => node.getBoundingClientRect().top)
```

Comment:
This would warrant a comment. And it’s not super clear to me why the `filter` is needed? wouldn’t there not be a `ref` in the set if it has no value?

Reply:
Good point, the filter is a holdout from when I hadn't added the unmount logic - I'll remove the filter and add a comment